core/src/main/java/org/apache/brooklyn/core/mgmt/BrooklynTags.java

```diff
@@ -59,6 +59,21 @@ public String getKind() {
         public String getContents() {
             return contents;
         }
+
+        @Override
+        public boolean equals(Object other) {
+            if (other == null || !(other instanceof NamedStringTag)) {
+                return false;
+            }
+
+            return kind != null && kind.equals(((NamedStringTag)other).kind)
```

Comment:
Thanks @aledsage ! I was looking for thing like `Objects.equal` but didn't find at the time.